kbs/src/api/src/http/resource.rs

```diff
@@ -123,13 +99,24 @@ pub(crate) async fn get_resource(
         .await
         .map_err(|e| Error::ReadSecretFailed(e.to_string()))?;
 
-    let jwe = jwe(pubkey, resource_byte)?;
-
-    let res = serde_json::to_string(&jwe).map_err(|e| Error::JWEFailed(e.to_string()))?;
-
-    Ok(HttpResponse::Ok()
-        .content_type("application/json")
-        .body(res))
+    // If public key is included in claims, use JWE to wrap resource byte,
+    // else check if HTTPS is enabled.
+    match get_pkey(claims) {
+        Ok(pubkey) => {
+            let jwe = jwe(pubkey, resource_byte)?;
+            let res = serde_json::to_string(&jwe).map_err(|e| Error::JWEFailed(e.to_string()))?;
+            Ok(HttpResponse::Ok()
+                .content_type("application/json")
+                .body(res))
+        }
+        Err(_) => {
+            if !https_enabled.enabled {
+                return Err(Error::InsecureResponse);
+            }
+            let res = URL_SAFE_NO_PAD.encode(resource_byte);
+            Ok(HttpResponse::Ok().body(res))
```

Comment:
I am confused by this PR. Isn't it fundamental that the secure channel is bound to the hw evidence? In other words, doesn't the KBS want to know that when it encrypts a value using a public key, it can only be decrypted inside the enclave that has the corresponding private key? I'm not sure it's valid to support secret delivery without this. What use case are you looking at?

Reply:
I am trying to get CDH to support calling AA to obtain a token from AS, and then using this token to access KBS to obtain resources.

Reply:
Here is the [PR](https://github.com/confidential-containers/guest-components/pull/465) for AA to add runtime data parameter. If we accept that, maybe this PR can be closed.